### DISCUSSION TITLE
Porting MicroPython to PixHawk (FMUv3)

### DISCUSSION THREAD

**Discussion by yopknopixx** [micropython/micropython#10134] [Hardware & Peripherals]
Hello all! I am interested in porting MicroPython to a **PixHawk** clone for a project I'm working on. Till now I have managed to flash firmware to the board, but I'm unable to verify whether it is working (I've tried USB REPL and each of the serial ports). I am using the `STM32F429DISC `target, which is quite similar to the target board.
Incidentally, I have also tried the same for the **Omnibus F4 V2** board by making slight changes to the `ADAFRUIT_F405_EXPRESS` target. With this board, I can see the status LED blinking (it is solid blue and turns off ~20-30 seconds after powering on). I assume this means that the firmware is being loaded properly and is working, but I cannot see a COM port for the USB REPL.
I would really appreciate any help I can get to get REPL working on either of these boards.
Thanks in advance!

**Comment by dlech** [comment]
Are you going to write flight code in Python?

**Reply by yopknopixx** [reply]
Yes. I am aware it isn't the best idea. But I've had decent success with it on the BlackPill. Besides, both boards are good platforms for building other robotics projects as well. Don't you think that having MicroPython support them would be a good idea?

**Comment by yopknopixx** [comment]
Update: I got it working on the Omnibus F4 by setting `MICROPY_HW_CLK_PLLM ` parameter to `8` instead of `12`. (I'm not entirely sure what it is or why it worked). I would love some help figuring that out and whether something similar will work on the Pixhawk.
But the board takes a long time to boot and there is no native filesystem. So, I will try to figure that out next.
